File: p001/p001_comprehension.py
# Henrik Grønbech; https://projecteuler.net/problem=1

# One-liner
def sum_multiples(upper, mult0, mult1):
    return sum((i for i in range(upper) if i % mult0 == 0 or i % mult1 == 0))


if __name__ == "__main__":
    print(sum_multiples(1000, 3, 5))

File: p001/p001_comprehension.jl
# Henrik Grønbech; https://projecteuler.net/problem=1

function sum_multiples(upper, mult1, mult2)
    return sum([i for i=1:upper-1 if i % mult1 == 0 || i % mult2 == 0])
end


println(sum_multiples(1000, 3, 5))
